content/scripts/build.js

```diff
@@ -145,6 +145,36 @@ function validateSlug(slug) {
   }
 }
 
+function repairUri(uri) {
+  // Returns a lowercase URI with common irregularities repaired.
+  uri = uri.trim().toLowerCase();
+  if (!uri.startsWith("/")) {
+    // Ensure the URI starts with a "/".
+    uri = "/" + uri;
+  }
+  // Remove redundant forward slashes, like "//".
+  uri = uri.replace(/\/{2,}/g, "/");
+  // Ensure the URI starts with a valid locale.
+  const maybeLocale = uri.split("/")[1];
+  if (!VALID_LOCALES.has(maybeLocale)) {
+    if (maybeLocale === "en") {
+      // Converts URI's like "/en/..." to "/en-us/...".
+      uri = uri.replace(`/${maybeLocale}`, "/en-us");
+    } else {
+      // Converts URI's like "/web/..." to "/en-us/web/...", or
+      // URI's like "/docs/..." to "/en-us/docs/...".
+      uri = "/en-us" + uri;
+    }
+  }
```

Comment:
@peterbe I like your plan from https://github.com/mdn/stumptown-renderer/issues/500#issuecomment-617873999, and then instead of having `repairUri` here we can have something like `verifyUri` that complains loudly when encountering URI's that aren't in `allTitles`.

Reply:
I should add one more comment here. We still need `repairUri` as part of the process of cleaning the URI's coming in from the macros (you'll see this as part of my KS renderer work), where the other part of that cleaning process is replacing the URI's of redirects with the URI of their final destination. Perhaps we should also implement a way to report all of the URI's that after cleaning still don't exist within `allTitles`. Right now, those are silently returning empty lists and objects. I didn't add that to my KS renderer work (although I did have some crude reporting in place while working) because I thought we needed to think through want we wanted to do in those cases.

Reply:
I *think* understand. Either way, we need to slowly transition from loosy-goosy forgiving to a strict and sensitive builder. How we do that will be a test of how good we are an engineering team. One thing at a time I guess!